sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -401,15 +400,38 @@ def _apply_shrinkage(self):
         for leaf in self.finalized_leaves:
             leaf.value *= self.shrinkage
 
-    def _initialize_root(self, gradients, hessians):
+    def _initialize_root(self):
         """Initialize root node and finalize it if needed."""
+        tic = time()
+        if self.interaction_cst is not None:
+            allowed_features = set().union(*self.interaction_cst)
+            allowed_features = np.fromiter(
+                allowed_features, dtype=np.uint32, count=len(allowed_features)
+            )
+            arbitrary_feature = allowed_features[0]
+        else:
+            allowed_features = None
+            arbitrary_feature = 0
+
+        # TreeNode init needs the total sum of gradients and hessians. Therefore, we
+        # first compute the histograms und then compute the total grad/hess on an
+        # arbitrary feature histogram. This way we replace a loop over n_samples by a
+        # loop over n_bins.
+        histograms = self.histogram_builder.compute_histograms_brute(
+            self.splitter.partition,  # =self.root.sample_indices
+            allowed_features,
+        )
+        self.total_compute_hist_time += time() - tic
+
+        tic = time()
         n_samples = self.X_binned.shape[0]
         depth = 0
-        sum_gradients = sum_parallel(gradients, self.n_threads)
+        histogram_array = np.asarray(histograms[arbitrary_feature])
```

Comment:
Is the `asarray` here required?

```suggestion
        histogram_array = histograms[arbitrary_feature]
```

Reply:
Without it I get the error
```
sum_gradients = histogram_array["sum_gradients"].sum()
                    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "<stringsource>", line 411, in View.MemoryView.memoryview.__getitem__
  File "<stringsource>", line 693, in View.MemoryView._unellipsify
```

Reply:
Ah I see. `histograms` is a memoryview at that point. I am okay with your implementation.